python/moltres_xs.py

```diff
@@ -0,0 +1,269 @@
+        self.get_xs()
+        self.fix_xs()
+
+    class t16_line:
+        def __init__(self, ind, multi, entry):
+            self.index = ind
+            self.multi_line_flag = multi
+            self.xs_entry = entry
+
+    def fix_xs(self):
+        for i in range(self.num_burn):
+            for j in range(self.num_uni):
+                for k in range(self.num_temps):
+                    for ii in range(self.num_groups):
+                        start = ii*self.num_groups
+                        stop = start + self.num_groups
+                        self.xs_lib[i][j][k]["REMXS"][ii] += np.sum(
+                            self.xs_lib[i][j][k]["GTRANSFXS"][start:stop])\
+                            - self.xs_lib[i][j][k]["GTRANSFXS"][start+ii]
+                        if self.xs_lib[i][j][k]["FISSE"][ii] != 0:
+                            self.xs_lib[i][j][k]["FISSE"][ii] = (
+                                self.xs_lib[i][j][k]["FISSE"][ii] /
+                                self.xs_lib[i][j][k]["FISSXS"][ii]
+                                )
+
+    def get_xs(self):
+        uni = []
+        L = 0
+        m = 0
+        n = 0
+        lam_temp = 0
+        beta_temp = 0
+        for k, line in enumerate(self.lines):
+            if 'Identifier' in line:
+                val = int(self.lines[k+1].split()[0])
+                if val not in uni:
+                    uni.extend([val])
+                m = uni.index(val)
+                self.xs_lib[L][m][n]['BETA_EFF']\
+                    .extend(beta_temp)
+                self.xs_lib[L][m][n]['DECAY_CONSTANT']\
+                    .extend(lam_temp)
+            if 'branch no.' in line:
+                index = line.find(',')
+                L = int(line[index-4:index])
+                n = int(line.split()[-1])
+            for key in self.catch.keys():
+                if key in line:
+                    if self.catch[key].multi_line_flag:
+                        if (key == 'Betas'):
+                            beta_temp = self.get_multi_line_values(k)
+                        elif(key == 'Lambdas'):
+                            lam_temp = self.get_multi_line_values(k)
+                        else:
+                            self.xs_lib[L][m][n][self.catch[key].xs_entry[0]]\
+                                .extend(
+                                    self.get_multi_line_values(k)
+                                    )
+                    else:
+                        for dex, xs in enumerate(self.catch[key].xs_entry):
+                            dex = self.catch[key].index[dex]
+                            self.xs_lib[L][m][n][xs]\
+                                .extend([self.get_values(k, dex)])
+
+    def get_values(self, k, index):
+        val = list(np.array(self.lines[k+1].split()).astype(float))
+        return val[index]
+
+    def get_multi_line_values(self, k):
+        values = []
+        while True:
```

Comment:
I recommend you do "for val in lines[k+1].split():" rather than an infinite loop that breaks when it runs out of stuff to process.

Reply:
The reason this is an infinite loop that breaks is because the number of lines that beta, lambda and the scattering cross sections are printed on is a function of the energy group size or the number of precursors. I could probably change it to get a certain number of entries and return but I am not sure how I would make the number of precursor groups flexible.

Reply:
I think this whole thing can be collapsed into way less code, and be made more clear. Check this out:

```python
def isFloat(s):
    try:
        float(s)
        return True
    except:
        return False
def get_multi_line_values(self, k):
    return [k for k in lines[k+1].split if isFloat(k)]
```

Reply:
@gridley this comment?

Reply:
This would only parse a single line it would not check for entries that span more than one line. I would argue this less clear, `k` was originally intended to be a line index, now it is acting as a line index and a value all in the same line.

Reply:
Ah, my mistake. In the list comprehension, you’d have to just change k to anything else, say, s, or, num.

Reply:
OK, I looked at this again, and now see what this does better. So yeah, actually, it's definitely good as-is. If you wanted to use list comprehensions here, you'd have to nest it. And that would be nasty.